viziphant/unitary_event_analysis.py

```diff
@@ -180,203 +150,227 @@ def plot_ue(data, Js_dict, sig_level, binsize, winsize, winstep,
+                         f"function multiple times for each hash at a time.")
+    for key in ['Js', 'n_emp', 'n_exp', 'rate_avg']:
+        Js_dict[key] = Js_dict[key].squeeze()
+    neurons_participated = ue.inverse_hash_from_pattern(pattern_hash,
+                                                        N=n_neurons).squeeze()
+
+    t_winpos = ue._winpos(t_start=t_start, t_stop=t_stop, win_size=win_size,
+                          win_step=win_step)
+    Js_sig = ue.jointJ(significance_level)
 
     # figure format
+    plot_params_user = plot_params
     plot_params = plot_params_default.copy()
     plot_params.update(plot_params_user)
-    if len(plot_params['unit_real_ids']) != N:
+    if plot_params['unit_real_ids'] is None:
+        plot_params['unit_real_ids'] = ['not specified'] * n_neurons
+    if len(plot_params['unit_real_ids']) != n_neurons:
         raise ValueError('length of unit_ids should be' +
                          'equal to number of neurons!')
     plt.rcParams.update({'font.size': plot_params['fsize']})
-    plt.rc('legend', fontsize=plot_params['fsize'])
-
-    num_row, num_col = 6, 1
     ls = '-'
     alpha = 0.5
-    plt.figure(1, figsize=plot_params['figsize'])
-    if 'suptitle' in plot_params.keys():
-        plt.suptitle("Trial aligned on " +
-                     plot_params['suptitle'], fontsize=20)
+
+    fig, axes = plt.subplots(nrows=6, sharex=True,
+                             figsize=plot_params['figsize'])
+    axes[5].get_shared_y_axes().join(axes[0], axes[2], axes[5])
+
+    for ax in (axes[0], axes[2], axes[5]):
+        for n in range(n_neurons):
+            for tr, data_tr in enumerate(spiketrains):
+                ax.plot(data_tr[n].rescale('ms').magnitude,
+                        np.full_like(data_tr[n].magnitude,
+                                     fill_value=n * n_trials + tr),
+                        '.', markersize=0.5, color='k')
+        for n in range(1, n_neurons):
+            # subtract 0.5 to separate the raster plots;
+            # otherwise, the line crosses the raster spikes
+            ax.axhline(n * n_trials - 0.5, lw=0.5, color='k')
+        ymax = max(ax.get_ylim()[1], 2 * n_trials - 0.5)
+        ax.set_ylim([-0.5, ymax])
+        ax.set_yticks([n_trials - 0.5, 2 * n_trials - 0.5])
+        ax.set_yticklabels([1, n_trials], fontsize=plot_params['fsize'])
+        ax.set_ylabel('Trial', fontsize=plot_params['fsize'])
+
+    for i, ax in enumerate(axes):
+        ax.set_xlim([t_winpos[0], t_winpos[-1] + win_size])
+        ax.text(-0.05, 1.1, string.ascii_uppercase[i],
+                transform=ax.transAxes, size=plot_params['fsize'] + 5,
+                weight='bold')
+        for key in plot_params['events'].keys():
+            for event_time in plot_params['events'][key]:
+                ax.axvline(event_time, ls=ls, color='r', lw=plot_params['lw'],
+                           alpha=alpha)
```

Comment:
In docstring events type is stated as `list` (see L104). Here it's a `dict`. In fact, it is a dict where values are lists.